src/test/java/org/truffleruby/integration/RubyBasicTest.java

```diff
@@ -0,0 +1,376 @@
+/*
+ * Copyright (c) 2017 Oracle and/or its affiliates. All rights reserved. This
+ * code is released under a tri EPL/GPL/LGPL license. You can use it,
+ * redistribute it and/or modify it under the terms of the:
+ *
+ * Eclipse Public License version 1.0
+ * GNU General Public License version 2
+ * GNU Lesser General Public License version 2.1
+ */
+package org.truffleruby.integration;
+
+import org.graalvm.options.OptionDescriptor;
+import org.graalvm.polyglot.Context;
+import org.graalvm.polyglot.Engine;
+import org.graalvm.polyglot.PolyglotException;
+import org.graalvm.polyglot.Source;
+import org.graalvm.polyglot.Value;
+import org.graalvm.polyglot.proxy.Proxy;
+import org.graalvm.polyglot.proxy.ProxyExecutable;
+import org.graalvm.polyglot.proxy.ProxyObject;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertSame;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+public class RubyBasicTest {
+
+    private Context context;
+
+    @Before
+    public void before() {
+        context = Context.create("ruby");
+    }
+
+    // ACCESS GUEST LANGUAGE DATASTRUCTUES
+
+    private Value eval(String code) {
+        return context.eval("ruby", code);
+    }
+
+    @Test
+    public void testReadArray() {
+        Value value = eval("[42,2,3]");
+        assertEquals(42, value.getArrayElement(0).asInt());
+    }
+
+    @Test
+    public void testIterateArray() {
+        Value value = eval("[42,43,44]");
+        for (int i = 0; i < value.getArraySize(); i++) {
```

Comment:
If `getArraySize()` returns 0, this tests nothing. Better assert it is 3 before iterating.